src/dependencies/connect.tsx

```diff
@@ -85,11 +85,13 @@ export default function connect(
       };
 
       render() {
+        const refProp = !!(BaseComponent.prototype && BaseComponent.prototype.isReactComponent) ? {ref: this.setWrappedInstance} : {};
```

Comment:
not that it's strictly relevant for this PR, but if we're supporting react 16 features we should probably move to `React.createRef` instead of a function ref?

Reply:
Code style nit: 

```
    const isStateful = !!(
      BaseComponent.prototype && BaseComponent.prototype.isReactComponent
    );
    return isStateful ? { ref: this.setWrappedInstance } : {};
```

seems a little bit prettier (pun intended)